source/common/network/connection_impl.cc

```diff
@@ -34,6 +34,18 @@ Address::InstanceConstSharedPtr getNullLocalAddress(const Address::Instance& add
 }
 } // namespace
 
+int ConnectionImplUtility::createSocket(Address::InstanceConstSharedPtr address,
+                                        const Address::InstanceConstSharedPtr source_address) {
+  const int fd = address->socket(Address::SocketType::Stream);
+  if (fd >= 0 && source_address != nullptr) {
+    int rc = source_address->bind(fd);
+    // TODO(alyssawilk) make this a non-fatal connect failure.
```

Comment:
Do you want to make it a RELEASE_ASSERT in the interim? Otherwise, we will have silent failure on release builds.